oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/measure/BanyanDBTopologyQueryDAO.java

```diff
@@ -123,72 +118,77 @@ List<Call.CallDetail> queryServiceRelation(Duration duration,
                                                                           String serverServiceId,
                                                                           Duration duration) throws IOException {
-        List<QueryBuilder<MeasureQuery>> queryBuilderList = buildInstanceRelationsQueries(
-            clientServiceId, serverServiceId);
-        return queryInstanceRelation(duration, queryBuilderList, DetectPoint.SERVER);
+        QueryBuilder<MeasureQuery> queryBuilder = buildInstanceRelationsQuery(
+                clientServiceId, serverServiceId);
+        return queryInstanceRelation(duration, queryBuilder, DetectPoint.SERVER);
     }
 
     @Override
     public List<Call.CallDetail> loadInstanceRelationDetectedAtClientSide(String clientServiceId,
                                                                           String serverServiceId,
                                                                           Duration duration) throws IOException {
-        List<QueryBuilder<MeasureQuery>> queryBuilderList = buildInstanceRelationsQueries(
-            clientServiceId, serverServiceId);
-        return queryInstanceRelation(duration, queryBuilderList, DetectPoint.CLIENT);
+        QueryBuilder<MeasureQuery> queryBuilder = buildInstanceRelationsQuery(
+                clientServiceId, serverServiceId);
+        return queryInstanceRelation(duration, queryBuilder, DetectPoint.CLIENT);
     }
 
-    private List<QueryBuilder<MeasureQuery>> buildInstanceRelationsQueries(String clientServiceId,
-                                                                           String serverServiceId) {
-        List<QueryBuilder<MeasureQuery>> queryBuilderList = new ArrayList<>(2);
-        queryBuilderList.add(new QueryBuilder<MeasureQuery>() {
+    private QueryBuilder<MeasureQuery> buildInstanceRelationsQuery(String clientServiceId,
+                                                                   String serverServiceId) {
+        return new QueryBuilder<MeasureQuery>() {
             @Override
             protected void apply(MeasureQuery query) {
-                query.and(eq(ServiceInstanceRelationServerSideMetrics.SOURCE_SERVICE_ID, clientServiceId))
-                     .and(eq(ServiceInstanceRelationServerSideMetrics.DEST_SERVICE_ID, serverServiceId));
-            }
-        });
+                List<AbstractCriteria> instanceRelationsQueryConditions = new ArrayList<>(2);
 
-        queryBuilderList.add(new QueryBuilder<MeasureQuery>() {
-            @Override
-            protected void apply(MeasureQuery query) {
-                query.and(eq(ServiceInstanceRelationServerSideMetrics.DEST_SERVICE_ID, clientServiceId))
-                     .and(eq(ServiceInstanceRelationServerSideMetrics.SOURCE_SERVICE_ID, serverServiceId));
+                instanceRelationsQueryConditions.add(
+                        // source_service_id = clientServiceId AND dest_service_id = serverServiceId
```

Comment:
What is this commented code about?

Reply:
The comment describes the query condition below